wordfreq/tokens.py

```diff
@@ -364,39 +186,66 @@ def tokenize(text, lang, include_punctuation=False, external_wordlist=False,
     If you end up seeing tokens that are entire phrases or sentences glued
     together, that probably means you passed in CJK text with the wrong
     language code.
+    """
+    # Use globals to load CJK tokenizers on demand, so that we can still run
+    # in environments that lack the CJK dependencies
+    global _mecab_tokenize, _jieba_tokenize
+
+    language = langcodes.get(lang)
+    info = get_language_info(language)
+    text = preprocess_text(text, language)
+
+    if info['tokenizer'] == 'mecab':
+        from wordfreq.mecab import mecab_tokenize as _mecab_tokenize
+        # Get just the language code out of the Language object, so we can
+        # use it to select a MeCab dictionary
+        tokens = _mecab_tokenize(text, language.language)
+        if not include_punctuation:
+            tokens = [token for token in tokens if not PUNCT_RE.match(token)]
+    elif info['tokenizer'] == 'jieba':
+        from wordfreq.chinese import jieba_tokenize as _jieba_tokenize
+        tokens = _jieba_tokenize(text, external_wordlist=external_wordlist)
+        if not include_punctuation:
+            tokens = [token for token in tokens if not PUNCT_RE.match(token)]
+    else:
+        # This is the default case where we use the regex tokenizer. First
+        # let's complain a bit if we ended up here because we don't have an
+        # appropriate tokenizer.
+        if info['tokenizer'] != 'regex':
+            if lang not in _WARNED_LANGUAGES:
```

Comment:
Nitpick: worth simplifying to `if info['tokenizer'] != 'regex' and lang not in _WARNED_LANGUAGES`?